AllReadyApp/Web-App/AllReady/Areas/Admin/Controllers/SkillAdminController.cs

```diff
@@ -0,0 +1,101 @@
+﻿using AllReady.Models;
+using AllReady.Security;
+using Microsoft.AspNet.Authorization;
+using Microsoft.AspNet.Mvc;
+using System.Linq;
+
+namespace AllReady.Areas.Admin.Controllers
+{
+    [Area("Admin")]
+    [Authorize("SiteAdmin")]
+    public class SkillController : Controller
+    {
+        private readonly IAllReadyDataAccess _dataAccess;
+
+        public SkillController(IAllReadyDataAccess dataAccess)
+        {
+            _dataAccess = dataAccess;
+        }
+
+        private ViewResult WithSkills(ViewResult view)
+        {
+            view.ViewData["Skills"] = _dataAccess.Skills;
+            return view;
+        }
+
+        public IActionResult Index()
+        {
+            return View(_dataAccess.Skills);
+        }
+
+        public IActionResult Create()
+        {
+            return WithSkills(View("Edit"));
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Create(Skill skill)
+        {
+            if (ModelState.IsValid)
+            {
+                _dataAccess.AddSkill(skill);
+                return RedirectToAction("Index", new { area = "Admin" });
+            }
+
+            return WithSkills(View("Edit", skill));
+        }
+
+        public IActionResult Edit(int? id)
+        {
+            if (id == null)
+            {
+                return new HttpStatusCodeResult(404);
+            }
+            Skill skill = _dataAccess.GetSkill((int)id);
+            if (skill == null)
+            {
+                return new HttpStatusCodeResult(404);
+            }
+
+            return WithSkills(View(skill));
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Edit(Skill skill)
+        {
+            if (ModelState.IsValid)
+            {
+                _dataAccess.UpdateSkill(skill);
+                return RedirectToAction("Index", new { area = "Admin" });
+            }
+            return WithSkills(View(skill));
+        }
+
+        public IActionResult Delete(int? id)
```

Comment:
It was preventing the 1st record from being inserted, since there isn't a FK back to an existing record.